src/map/party.c

```diff
@@ -300,6 +300,18 @@ int party_recv_info(const struct party *sp, int char_id)
 			if (i == MAX_PARTY)
 				added[added_count++] = member_id;
 		}
+		
+		ARR_FIND(0, MAX_PARTY, j, p->party.member[j].leader == 1);
+		if (j == MAX_PARTY) {
+			// Leader has changed
+			int k;
+			ARR_FIND(0, MAX_PARTY, k, sp->member[k].leader == 1);
+			if (k < MAX_PARTY) {
+				clif->PartyLeaderChanged(map->id2sd(sp->member[k].account_id), 0, sp->member[k].account_id);
```

Comment:
Is there any reason to lookup `map->id2sd(sp->member[k].account_id)`, when the sd pointer is already cached in `p->data[k].sd`?
But either way, `map->id2sd()` may return a `NULL` pointer if the player is offline. The correct behavior here would be to find the first available sd in the party, and then use that one to pass to `clif->PartyLeaderChanged()`. The second argument should be the old account ID, which is still available at the beginning of this function, so you could retrieve it.
If you move this to the `while (removed_count > 0)` block below, and check for the removed member's `.leader`, you can even save an `ARR_FIND()`, and only do this when the leader is actually getting removed.